lib/js/src/manager/screen/_SoftButtonReplaceOperation.js

```diff
@@ -0,0 +1,307 @@
+        }
+        return response.getSuccess();
+    }
+
+    /**
+     * Uploads soft buttons representing the current states of the button objects
+     * @private
+     * @returns {Promise} - Resolves to whether the operation is successful
+     */
+    async _sendCurrentStateSoftButtons () {
+        if (this.getState() === _Task.CANCELED) {
+            return false;
+        }
+        console.log('SoftButtonTransitionOperation - Preparing to send full soft buttons');
+        const softButtons = [];
+
+        for (const softButtonObject of this._softButtonObjects) {
+            softButtons.push(softButtonObject.getCurrentStateSoftButton());
+        }
+
+        if (this._lifecycleManager === null) {
+            console.error('SoftButtonTransitionOperation: LifecycleManager is null');
+            return false;
+        }
+
+        const show = new Show()
+            .setMainField1(this._currentMainField1)
+            .setSoftButtons(softButtons);
+
+        const response = await this._lifecycleManager.sendRpcResolve(show);
+        if (response.getSuccess()) {
+            console.log('SoftButtonTransitionOperation - Finished sending full soft buttons');
+        } else {
+            console.warn('SoftButtonTransitionOperation - Failed to update soft buttons');
+        }
+        return response.getSuccess();
+    }
+
+    /**
+     * Upload all soft button images, the initial state images first, then the other states. We need to send updates when the initial state is ready.
+     * @private
+     * @returns {Promise} - Resolves to whether the operation is successful
+     */
+    async _uploadInitialStateImages () {
+        if (this.getState() === _Task.CANCELED) {
+            return false;
+        }
+        const initialStatesToBeUploaded = [];
+
+        for (const softButtonObject of this._softButtonObjects) {
+            const softButtonState = softButtonObject.getCurrentState();
+            if (softButtonState !== null && this._fileManager.fileNeedsUpload(softButtonState.getArtwork())) {
+                initialStatesToBeUploaded.push(softButtonState.getArtwork());
+            }
+        }
+
+        if (initialStatesToBeUploaded.length === 0) {
+            console.log('SoftButtonTransitionOperation: No initial state artworks to upload');
+            return false;
+        }
+
+        console.log('SoftButtonTransitionOperation: Uploading soft button initial artworks');
+        if (this._fileManager === null) {
+            return false;
+        }
+
+        const successes = await this._fileManager.uploadArtworks(initialStatesToBeUploaded);
+        if (successes.includes(false)) {
+            console.error('SoftButtonTransitionOperation: Error uploading soft button artworks');
+        } else {
+            console.log('SoftButtonTransitionOperation: Soft button initial state artworks uploaded');
+        }
+
+        return !successes.includes(false);
```

Comment:
It doesn't impact the manager logic at the moment but the value that Java passes in to the completion listener here isn't whether the operation was successful, it's whether the operation was **not** canceled while running.